util/annotations/annotations.go

```diff
@@ -67,17 +69,24 @@ func (a Annotations) AsErrors() []error {
 }
 
 // AsStrings is a convenience function to return the annotations map as a slice
-// of strings.
+// of strings. This is only used in the API response, so we limit the number of
+// strings here.
 func (a Annotations) AsStrings(query string) []string {
 	arr := make([]string, 0, len(a))
 	for _, err := range a {
+		if len(arr) >= maxErrsInAPI {
+			break
+		}
 		anErr, ok := err.(annoErr)
 		if ok {
 			anErr.Query = query
 			err = anErr
 		}
 		arr = append(arr, err.Error())
 	}
+	if len(a) > maxErrsInAPI {
+		arr = append(arr, fmt.Sprintf("%d more PromQL warnings/annotations", len(a)-maxErrsInAPI))
```

Comment:
is it only promql? this is a generic package

Reply:
It can include storage as well, so this has already been changed to the generic "%d more annotations omitted" as per Bjorn's recommendation.